### PR TITLE
Improve the help and output of the `CONFIG` command

### PR DESCRIPTION
**Pull request by johnnovak**
Addresses my complaints raised in https://github.com/dosbox-staging/dosbox-staging/issues/2669 plus gives the output of the `CONFIG` command some much needed love.

This is how it looks like now:

image
image

image

image

### CODE REVIEW

File: src/misc/programs.cpp [hunk 1]
```diff
@@ -891,76 +878,129 @@ void PROGRAMS_Destroy([[maybe_unused]] Section* sec)
...
+	        "\n"
+	        "  -wcp [color=white]FILENAME     [/reset]Writes the config to the current working directory as\n"
+	        "                    'dosbox.conf', or as the specified filename.\n"
+	        "  -wcd              Writes the config to the default global config file.\n"
+	        "\n"
+	        "  -writelang [color=white]FILENAME[/reset]\n"
+	        "  -wl [color=white]FILENAME      [/reset]Writes the current language strings to [color=white]FILENAME [/reset]in the\n"
+	        "                    current working directory.\n"
+	        "\n"
+	        "  -r [color=cyan][PROPERTY1=VALUE1][ PROPERTY2=VALUE2]...[/reset]\n"
```
Comment (FeralChild64) [inline src/misc/programs.cpp:923]: `[ PROPERTY2=VALUE2]` - space is not needed here. I suggest:
`[PROPERTY1=VALUE1][ PROPERTY2=VALUE2]...` -> `[PROPERTY1=VALUE1 [PROPERTY2=VALUE2 ...]]`

File: src/misc/programs.cpp [hunk 2]
```diff
@@ -891,76 +878,129 @@ void PROGRAMS_Destroy([[maybe_unused]] Section* sec)
...
-	        "-axclear clears the autoexec section.\n"
-	        "-axadd [line] adds a line to the autoexec section.\n"
-	        "-axtype prints the content of the autoexec section.\n"
-	        "-securemode switches to secure mode.\n"
-	        "-avistart starts AVI recording.\n"
-	        "-avistop stops AVI recording.\n"
-	        "-startmapper starts the keymapper.\n"
-	        "-get \"section property\" returns the value of the property.\n"
-	        "-set \"section property=value\" sets the value.\n");
+	        "Performs configuration management tasks and other miscellaneous actions.\n"
```
Comment (FeralChild64) [inline src/misc/programs.cpp:902]: Description line is too long for `help /all` command:

description

File: src/hardware/mixer.cpp [hunk 3]
```diff
@@ -2377,7 +2377,7 @@ class MIXER final : public Program {
 	void AddMessages()
 	{
 		MSG_Add("SHELL_CMD_MIXER_HELP_LONG",
-		        "Displays or changes the sound mixer settings.\n"
+		        "\nDisplays or changes the sound mixer settings.\n"
```
Comment (FeralChild64) [inline src/hardware/mixer.cpp:2380]: Do not start help with new line, this breaks description in `help /all` command.

File: src/misc/programs.cpp (hunk 1 above)
Comment (johnnovak) [reply src/misc/programs.cpp:923]: Sure, that's better.

File: src/misc/programs.cpp (hunk 2 above)
Comment (johnnovak) [reply src/misc/programs.cpp:902]: Ah, forgot about that!

File: src/hardware/mixer.cpp (hunk 3 above)
Comment (johnnovak) [reply src/hardware/mixer.cpp:2380]: Ok.

File: src/misc/programs.cpp [hunk 4]
```diff
@@ -891,76 +878,129 @@ void PROGRAMS_Destroy([[maybe_unused]] Section* sec)
...
+	        "  -writeconf\n"
+	        "  -wc               Writes the config to the primary config file.\n"
+	        "\n"
+	        "  -writeconf [color=white]FILENAME[/reset]\n"
+	        "  -wc [color=white]PATH          [/reset]Writes the config to [color=white]PATH[/reset] relative to the config directory,\n"
+	        "                    or to an absolute path.\n"
+	        "\n"
+	        "  -wcp [color=white]FILENAME     [/reset]Writes the config to the current working directory as\n"
+	        "                    'dosbox.conf', or as the specified filename.\n"
+	        "  -wcd              Writes the config to the default global config file.\n"
```
[the remark this answers is not in the document]
Comment (johnnovak) [reply src/misc/programs.cpp:917]: I can take your word for it and change it to that :) I deliberately did not want to spend time on untangling what the different config write options exactly do (I find them confusing, and the old help wasn't of much help...) That can be done separately if someone has the appetite for it.

**Comment by johnnovak** [comment]
Okay, I've made some further small enhancements here and there, and have simplified and improved the "write config" functionality. It was all quite horrible and badly designed without any concern for good user-experience principles whatsoever. I spent 30 minutes today understanding it and testing it out, and even when I knew how it worked, it was still all quite counterintuitive... It just had to be simplified, and I remember I was always quite unsure what it was _exactly_ doing whenever I used it before. 

This is how the help looks now; it should be straightforward to use and simple to understand. I've also improved the various config related error and status messages, plus added error messages for wrong invocations (previously, if you for example provided too many arguments, nothing happened... no error, no output, no nothing... 🤦🏻 😬 )

image

File: src/misc/programs.cpp (hunk 4 above)
[the remark this answers is not in the document]
Comment (johnnovak) [reply src/misc/programs.cpp]: Okay, that someone was me later today as this badly designed feature made me quite upset, so I had to hack'n'slash it... 

> I think the main difference with -wc and -wcp is that those two save you current running config, while -wcd saves the default config instead, ignoring any config changes that were made.

No, it wasn't possible to do that; that's also clear if you read the code. See my last comment below on how I simplified and improved this.

**Comment by rderooy** [comment]
Looks good to me @johnnovak !

**Pull request merged**